profile/profile.js

```diff
@@ -98,12 +104,67 @@ WDN.initializePlugin('modal', [function() {
 				}
 			});
 		}
+
+
+
+		// Maps a roleId -> siteId -> maximumNumber of that role allowed for that site (shift can override)
+		const siteRoleLimitsMap = new Map();
+		// Maps a roleId -> shiftId -> maximumNumber of that role allowed for that shift
+		const shiftRoleLimitsMap = new Map();
+
+		function loadRoleLimits() {
+			$.ajax({
+				url: '/server/api/roles/limits/getAll.php',
+				type: 'GET',
+				dataType: 'JSON',
+				cache: false,
+				success: function(response) {
+					for (let i = 0; i < response.length; i++) {
+						const roleLimit = response[i];
+
+						const isSiteRoleLimitEntry = roleLimit.shiftId == null;
+						if (isSiteRoleLimitEntry) {
+							addToSiteRoleLimitsMap(roleLimit);
+						} else {
+							addToShiftRoleLimitsMap(roleLimit);
+						}
+					}
+				},
+				error: function(response) {
+					alert('Unable to load role limits. Please refresh the page in a few minutes.');
+				}
+			});
+		}
+
+		function addToSiteRoleLimitsMap(siteRoleLimit) {
+			const hasRoleIdAlready = siteRoleLimitsMap.has(siteRoleLimit.roleId);
+			if (!hasRoleIdAlready) {
+				siteRoleLimitsMap.set(siteRoleLimit.roleId, new Map());
```

Comment:
Yah, if you saw the page it might make a little more sense. But the idea is when people sign up for a shift they have to select a site, date, time, and role. The times are determined by which date they select, and the dates are determined by which site they select. And now the role is determined by which site/date/time they select.